src/PartnerRegistrar.ts

```diff
@@ -392,7 +395,7 @@ export class PartnerRegistrar {
    * @param amount the amount for the name registration
    * @param addr the address to set for the name resolution
    */
-  commitAndRegister (label: string, owner: string, duration: BigNumber, amount: BigNumber, addr?: string): Promise<boolean> {
+  commitAndRegister (label: string, owner: string, duration: BigNumber, amount: BigNumber, addr?: string): Promise<string> {
```

Comment:
Maybe in this case we can both return the commit hash and the register hash? Either as an object or an array.

```suggestion
  commitAndRegister (label: string, owner: string, duration: BigNumber, amount: BigNumber, addr?: string): Promise<[string, string]> {
```

Reply:
Great suggestion!